test/testutils/set_scene_test.py

```diff
@@ -372,3 +416,182 @@ def objectScaleSet():
+	def meshTextureMapping():
+		testutils.set_scene_data({
+			'obj': { 'mesh': 'test/testutils/pyramid.dae' }
+		})
+
+		mesh = bpy.data.objects.get('obj').to_mesh()
+
+		assert_that(mesh.uv_layers, has_length(1), 'Mesh UV Layer set up')
+
+		uvs = mesh.uv_layers[0].data
+
+		assert_that(uvs, has_length(16), 'Read in expected number of UV Loops')
+		assert_that(uvs[ 0].uv, equal_to(Vector((0.5, 0.5))), 'UV  0')
+		assert_that(uvs[ 1].uv, equal_to(Vector((0.0, 0.0))), 'UV  1')
+		assert_that(uvs[ 2].uv, equal_to(Vector((1.0, 0.0))), 'UV  2')
+		assert_that(uvs[ 3].uv, equal_to(Vector((0.5, 0.5))), 'UV  3')
+		assert_that(uvs[ 4].uv, equal_to(Vector((1.0, 0.0))), 'UV  4')
+		assert_that(uvs[ 5].uv, equal_to(Vector((1.0, 1.0))), 'UV  5')
+		assert_that(uvs[ 6].uv, equal_to(Vector((0.0, 1.0))), 'UV  6')
+		assert_that(uvs[ 7].uv, equal_to(Vector((0.5, 0.5))), 'UV  7')
+		assert_that(uvs[ 8].uv, equal_to(Vector((1.0, 1.0))), 'UV  8')
+		assert_that(uvs[ 9].uv, equal_to(Vector((0.0, 1.0))), 'UV  9')
+		assert_that(uvs[10].uv, equal_to(Vector((0.0, 0.0))), 'UV 10')
+		assert_that(uvs[11].uv, equal_to(Vector((0.5, 0.5))), 'UV 11')
+		assert_that(uvs[12].uv, equal_to(Vector((1.0, 1.0))), 'UV 12')
+		assert_that(uvs[13].uv, equal_to(Vector((1.0, 0.0))), 'UV 13')
+		assert_that(uvs[14].uv, equal_to(Vector((0.0, 0.0))), 'UV 14')
+		assert_that(uvs[15].uv, equal_to(Vector((0.0, 1.0))), 'UV 15')
+
+	@it('Loading materials from mesh file')
+	def meshMaterials():
+		testutils.set_scene_data({
+			'obj': { 'mesh': 'test/testutils/pyramid.dae' }
+		})
+
+		mesh = bpy.data.meshes.get('Pyramid')
+		mat = bpy.data.materials.get('Pyramid')
+
+		assert_that(mesh.materials,
+			has_entry('Pyramid', mat),
+			'Mesh has material set')
+
+		assert_that(mat.diffuse_color,
+			contains(1, 0, 0, 1),
+			'Material diffuse color set')
+
+		assert_that(mat.name, equal_to('Pyramid'), 'Material name set')
+
+	@it('Loading material assignments from mesh file')
+	def meshFaceMaterials():
+		testutils.set_scene_data({
+			'obj': { 'mesh': 'test/testutils/pyramid.dae' }
+		})
+
+		mat1 = bpy.data.materials.get('Pyramid')
+		mat2 = bpy.data.materials.get('Second')
+
+		assert_that(mat1, not_none(), 'First material exists')
+		assert_that(mat2, not_none(), 'First material exists')
+
+		mats = bpy.data.materials
+		p = bpy.data.objects.get('obj').to_mesh().polygons
+
+		# This is the only way I can find to link a polygon's material to a
+		# Material object. This seems weirdly incongruent with the rest of
+		# Blender's API, so maybe I'm missing something?
+		assert_that(mats[p[0].material_index], equal_to(mat1), 'Poly 0 Pyramid material')
+		assert_that(mats[p[1].material_index], equal_to(mat1), 'Poly 1 Pyramid material')
+		assert_that(mats[p[2].material_index], equal_to(mat1), 'Poly 2 Pyramid material')
+		assert_that(mats[p[3].material_index], equal_to(mat1), 'Poly 3 Pyramid material')
+		assert_that(mats[p[4].material_index], equal_to(mat2), 'Poly 4 Second material')
```

Comment:
I'm not sure either. But this would explain why I was having trouble applying materials last time I tried to. I did not at all expect to need to do it like this.